lbcapi/api.py

```diff
@@ -86,12 +92,13 @@ def call(self, method, url, params=None, stream=False, files=None):
                 # GET method
                 else:
                     api_request = requests.Request('GET', self.server + url, params=params).prepare()
-                    params_encoded = urlparse.urlparse(api_request.url).query
+                    params_encoded = urlparse(api_request.url).query
 
                 # Calculate signature
-                message = nonce + self.hmac_key + str(url)
+                # .encode('ascii') ensures a bytestring on Python 2.7 and 3.x
+                message = nonce + self.hmac_key + url.encode('ascii')
                 if params_encoded:
-                    message += str(params_encoded)
+                    message += str(params_encoded).encode('ascii')
```

Comment:
Looks good, except here `params_encoded` may contain binary data when sending files. If that is the case it will raise `UnicodeDecodeError` at least on Python 2.7.

If you want to fix this problem, here is how to test it. In this example an image is sent to contact chat.

```
    with open('/tmp/test_image.jpg', 'rb') as file:
        file_bytes = file.read()
    files = {
        'document': ('test_image.jpg', file_bytes, 'image/jpeg'),
    }
    make_api_call(request, 'POST', '/api/contact_message_post/XXXXXXXX/', {}, files=files)
```

I can make the actual testing, but if you want to do it and don't have any good contacts to test with, you can contact my testing ad: https://localbitcoins.com/ad/395251/purchase-bitcoin-paypal-finland